### PR TITLE
throw no database select error msg

### PR DESCRIPTION
**Pull request by jianwan0214**
## What type of PR is this?

- [ ] API-change
- [x] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #4865 

## What this PR does / why we need it:
throw no database select error msg

### CODE REVIEW

File: pkg/sql/compile/ddl.go [hunk 1]
```diff
@@ -71,6 +71,9 @@ func (s *Scope) CreateTable(c *Compile) error {
 	}
 	dbSource, err := c.e.Database(c.ctx, dbName, c.proc.TxnOperator)
 	if err != nil {
+		if dbName == "" {
+			return errors.New("", "No database selected")
```
Comment (fengttt) [inline pkg/sql/compile/ddl.go:75]: You should use moerr now.

Comment (jianwan0214) [reply pkg/sql/compile/ddl.go:75]: ok

File: pkg/sql/compile/ddl.go [hunk 2]
```diff
@@ -71,6 +72,9 @@ func (s *Scope) CreateTable(c *Compile) error {
 	}
 	dbSource, err := c.e.Database(c.ctx, dbName, c.proc.TxnOperator)
 	if err != nil {
+		if dbName == "" {
+			return moerr.NewError(moerr.BAD_CONFIGURATION, "No database selected")
```
Comment (iamlinjunhong) [inline pkg/sql/compile/ddl.go:76]: use moerr.New(moerr.ER_NO_DB_ERROR)

**Review by nnsgmsone — APPROVED** [review]

File: pkg/sql/compile/ddl.go (hunk 2 above)
Comment (jianwan0214) [reply pkg/sql/compile/ddl.go:76]: ok

**Comment by iamlinjunhong** [comment]
Why merged?

**Pull request merged**